src/util/apple/scheduler.hpp

```diff
@@ -0,0 +1,204 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <realm/util/to_string.hpp>
+
+#include <atomic>
+#include <dispatch/dispatch.h>
+#include <objc/runtime.h>
+#include <pthread.h>
+
```

Comment:
Don't we need include guards?

Reply:
These aren't really intended to be header files. They're just the platform-specific parts of scheduler.cpp split into separate files for organizational reasons, but compiled as part of scheduler.cpp to avoid complicating the build system (both here and in all the SDKs consuming it). Including them anywhere else would result in duplicated symbols.

Reply:
ah I didn't catch that thanks, sounds good